lib/block-supports/typography.php

```diff
@@ -80,6 +80,21 @@ function gutenberg_apply_typography_support( $block_type, $block_attributes ) {
 		return array();
 	}
 
+	return gutenberg_typography_generate_css_classes_and_styles( $typography_supports, $block_attributes );
+}
+
+/**
+ * Generates all the typography block support related CSS classes and inline
+ * styles. This allows block supports config to be passed manually such that
+ * individual support features can be opted out of while still easily generating
+ * classes and styles for other features.
+ *
+ * @param array $typography_supports Typography block support flags.
+ * @param array $block_attributes    Block's attributes.
+ *
+ * @return array Typography CSS classes and inline styles.
+ */
+function gutenberg_typography_generate_css_classes_and_styles( $typography_supports, $block_attributes ) {
```

Comment:
I'm not sure if moving the css class and style generation to a function in this scope and using that is a good idea. I didn't see a lot of options otherwise in terms of not duplicating that logic in any dynamic block that skips serialization of block support styles but only really wishes to do so for a single feature, i.e. text decoration in this case.

@oandregal Do you have any suggestions or thoughts on this?

Reply:
If I read this correctly: we want to do this to have a helper function and avoid having to recreate all this code in navigation. If we were able to skip serialization for a single property it'd be simpler because the other properties would still be rendered by the block support function.

I think we should do this if it's helpful.

The only bit of feedback I have is about this function's signature. Once we have a function in WordPress core we can't remove it. So, at first sight, I'd make this the same as `gutenberg_apply_typography_support`: pass `$block_type` and `$block_attlributes`. The rationale would be that's more future proof: we don't know what we want to do in the future with this utility and perhaps we want to access the block name, the block style variations, etc. It makes it slightly more involved for your use case than passing a tweaked typography supports, but I hope that's not too much. It's in line with what we do in the client as well. What do you think of this?

Reply:
> If we were able to skip serialization for a single property it'd be simpler because the other properties would still be rendered by the block support function.
> 
> I think we should do this if it's helpful.

Agreed. As we attempt to adopt block supports for more complicated blocks I can see us running into this more often. 

I'll try my hand at putting together a PR to allow for this.

> The only bit of feedback I have is about this function's signature. Once we have a function in WordPress core we can't remove it. So, at first sight, I'd make this the same as gutenberg_apply_typography_support: pass $block_type and $block_attlributes. 

Good point. Thanks for the warning. 👍 

> It makes it slightly more involved for your use case than passing a tweaked typography supports, but I hope that's not too much. It's in line with what we do in the client as well. What do you think of this?

I don't think it's too much at all, better than restricting our ability to handle changes in future. I'll see how quickly the PR to allow skipping serialization of individual features comes along, then revisit this as required.